x-pack/plugins/observability_solution/observability_ai_assistant_app/public/functions/visualize_esql.tsx

```diff
@@ -238,88 +245,170 @@ export function VisualizeESQL({
+                content={i18n.translate('xpack.observabilityAiAssistant.lensESQLFunction.edit', {
+                  defaultMessage: 'Edit visualization',
                 })}
               >
                 <EuiButtonIcon
                   size="xs"
-                  iconType="save"
-                  onClick={() => setIsSaveModalOpen(true)}
-                  data-test-subj="observabilityAiAssistantLensESQLSaveButton"
+                  iconType="pencil"
+                  onClick={() => {
+                    chatFlyoutSecondSlotHandler?.setVisibility?.(true);
+                    if (triggerOptions) {
+                      uiActions.getTrigger('IN_APP_EMBEDDABLE_EDIT_TRIGGER').exec(triggerOptions);
+                    }
+                  }}
+                  data-test-subj="observabilityAiAssistantLensESQLEditButton"
                   aria-label={i18n.translate(
-                    'xpack.observabilityAiAssistant.lensESQLFunction.save',
+                    'xpack.observabilityAiAssistant.lensESQLFunction.edit',
                     {
-                      defaultMessage: 'Save visualization',
+                      defaultMessage: 'Edit visualization',
                     }
                   )}
                 />
               </EuiToolTip>
+              <EuiFlexItem grow={false}>
+                <EuiToolTip
+                  content={i18n.translate('xpack.observabilityAiAssistant.lensESQLFunction.save', {
+                    defaultMessage: 'Save visualization',
+                  })}
+                >
+                  <EuiButtonIcon
+                    size="xs"
+                    iconType="save"
+                    onClick={() => setIsSaveModalOpen(true)}
+                    data-test-subj="observabilityAiAssistantLensESQLSaveButton"
+                    aria-label={i18n.translate(
+                      'xpack.observabilityAiAssistant.lensESQLFunction.save',
+                      {
+                        defaultMessage: 'Save visualization',
+                      }
+                    )}
+                  />
+                </EuiToolTip>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </EuiFlexItem>
+          {isTableVisible && (
+            <EuiFlexItem data-test-subj="observabilityAiAssistantESQLDataGrid">
+              <ESQLDataGrid
+                rows={rows}
+                columns={columns}
+                dataView={dataViewAsync.value}
+                query={{ esql: query }}
+                flyoutType="overlay"
+                isTableView
+              />
+            </EuiFlexItem>
+          )}
+          {!isTableVisible && (
+            <EuiFlexItem data-test-subj="observabilityAiAssistantESQLLensChart">
+              <lens.EmbeddableComponent
+                {...lensInput}
+                style={{
+                  height: 240,
+                }}
+                onLoad={onLoad}
+              />
             </EuiFlexItem>
```

Comment:
can we do something like `<EuiFlexItem ...>{ isTableVisible ? <ESQLDataGrid/> : <lens.EmbeddableComponent/> }</EuiFlexItem>`? I'm fine with using one `data-test-subj` if that also works for you

Reply:
Of course we can! 👌

Reply:
Done https://github.com/elastic/kibana/pull/184106/commits/da816e8d9bf64e4a93a7ede44f078ff832598f04 I kept different data subj for my unit tests